openhtf/core/phase_executor.py

```diff
@@ -251,6 +261,13 @@ def stop(self):
     It will raise a ThreadTerminationError, which will cause the test to stop
     executing and terminate with an ERROR state.
     """
+    self.test_state.logger.debug('Stopping PhaseExecutor')
     current_phase_thread = self._current_phase_thread
     if current_phase_thread:
-      current_phase_thread.kill()
+      kill_start = time.time()
+      current_phase_thread.stop()
+      while current_phase_thread.is_alive():
+        time.sleep(.1)
+      self.test_state.logger.debug('Took %.3f seconds to kill %s',
+          time.time() - kill_start,
+          current_phase_thread.name)
```

Comment:
The approach I would have taken is to move the whole teardown function stuff up a level to the `TestExecutor`. It's already in charge of `test_start`, and has a perfect singular place to ensure the teardown gets called (the finally clause in `_execute_test_phases`, or a new try-finally in `_thread_proc` on L144) versus the 3 places it's mentioned here.

It seems to decrease complexity to me, what do you think?